### PR TITLE
Prefer unittest.mock before mock

### PR DESCRIPTION
**Pull request by mimi1vx**
From Python 3.3 is mock part of python standard library in unittest namespace

- [x] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
good reminder, needing mock is insane - good tests should do without

File: testing/test_mark.py [hunk 1]
```diff
@@ -1,7 +1,11 @@
 from __future__ import absolute_import, division, print_function
 import os
 import sys
-import mock
+
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
```
Comment (asottile) [inline testing/test_mark.py]: there are occasions where this is undesirable, for instance I'd prefer to use the backported `mock` (>=2) over the stdlib in python3.4:

```console
$ python3.4 -c 'from unittest import mock; mock.Mock().assert_wat()'
$ python3.4 -c 'import mock; mock.Mock().assert_wat()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.4/dist-packages/mock/mock.py", line 703, in __getattr__
    raise AttributeError(name)
AttributeError: assert_wat
```

Comment (asottile) [reply testing/test_mark.py]: I'm actually more of a +0/-0 on this change -- this is part of the test suite _for pytest_ and not part of the public package.  pytest also hard codes a test dependency [on mock](https://github.com/pytest-dev/pytest/blob/50a0d4fa95b3f6a25e650cd977307aebf5507d65/tox.ini#L25) so the import should never fail under normal circumstances

Comment (nicoddemus) [reply testing/test_mark.py]: I'm with @asottile, but @RonnyPfannschmidt seems to be +1 about it.

**Comment by mimi1vx** [comment]
@asottile @nicoddemus  so I changed logic to try import first upstream `mock` and if this fail stdlib unittest.mock

**Pull request merged**